src/BulkAppointments/BulkAppointmentsContainer.js

```diff
@@ -43,69 +39,96 @@ class BulkAppointmentsContainer extends Component {
     titleDispatch('Whereabouts')
   }
 
-  async onFileInputChanged(event) {
-    const { handleError, resetErrorDispatch } = this.props
-    resetErrorDispatch()
-
-    try {
-      event.preventDefault()
-      const formData = createFileFormDataFromEvent(event)
-
-      const response = await axios.post('/api/appointments/upload-offenders', formData, {
-        headers: {
-          'Content-Type': 'multipart/form-data',
-        },
-      })
-      this.setState({
-        offenders: response.data,
-      })
-    } catch (error) {
-      handleError(error)
-    }
-  }
-
-  BackToAppointmentDetails() {
-    const { appointmentTypes, appointment } = this.state
-    this.setState({
-      appointmentTypes,
-      appointment,
-      currentStep: 0,
-    })
+  previousStep() {
+    this.setState(state => ({
+      ...state,
+      currentStep: state.currentStep - 1,
+    }))
   }
 
   render() {
-    const { agencyId } = this.props
-    const { offenders, currentStep } = this.state
+    const { agencyId, handleError, resetErrorDispatch } = this.props
+    const { complete, offenders, currentStep, appointment } = this.state
 
-    const steps = [
-      <WithDataSource
-        request={{ url: '/api/bulk-appointments/view-model', params: { agencyId } }}
-        render={({ data, error }) => (
-          <AppointmentForm
-            now={moment()}
-            appointmentTypes={data.appointmentTypes}
-            error={error}
-            locationTypes={data.locationTypes}
-            trySubmit={values =>
-              this.setState(state => ({
-                ...state,
-                currentStep: currentStep + 1,
-                appointment: values,
-              }))
-            }
-          />
-        )}
-      />,
-
-      <div>
-        <BackLink onClick={() => this.BackToAppointmentDetails()}> Back</BackLink>
-        <ChooseOffenders onFileInputChanged={event => this.onFileInputChanged(event)} />
-        <OffenderTable offenders={offenders} />
-      </div>,
-    ]
     return (
-      <Page title="Add bulk appointments">
-        <div>{steps[currentStep]}</div>
+      <Page title="Add bulk appointments" alwaysRender>
+        {complete && (
+          <div>
+            <H3> Appointments have been successfully created. </H3>
+            <div>
+              <Button
+                onClick={() =>
+                  this.setState({
```

Comment:
I think its good practice to keep these function declarations outside the render() method - so they only get created once.